File: worklist-ui/src/main/java/com/gcgenome/lims/client/domain/PanelEvent.java
package com.gcgenome.lims.client.domain;

import lombok.Builder;
import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Accessors(fluent=true)
@Builder
public class PanelEvent<T> {
    private String id;
    private String type;
    private T param;
}

File: domain/src/main/kotlin/com/gcgenome/lims/domain/event/PanelEvent.kt
package com.gcgenome.lims.domain.event

import java.io.Serializable
import java.util.*

interface PanelEvent<T: Serializable, N: PanelEvent<T, N>>: Serializable {
    fun id(): UUID
    fun type(): Type
    fun param(): T
    enum class Type {
        UPDATE_SERIAL, UPDATE_REQUEST_SEQUENCE, UPDATE_SEQUENCING_INDEX, UPDATE_BATCH_SEQUENCING, UPDATE_SEQUENCING_ITEM
    }
}